### DISCUSSION TITLE
New Bevy LDtk Plugin

### DISCUSSION THREAD

**Discussion by zicklag** [bevyengine/bevy#1324] [Show and tell]
Hey everybody, I've just published a new Bevy plugin for loading [LDtk](https://ldtk.io) maps in Bevy: [bevy_ldtk](https://github.com/katharostech/bevy_ldtk)!

[![Screenshot from 2021-01-26 15-20-00](https://user-images.githubusercontent.com/25393315/105909406-23f3cc00-5fed-11eb-9071-402fa9ede885.png)](https://github.com/katharostech/bevy_ldtk)

The plugin features what I think is an efficient renderer that only uses a single Quad mesh for every map layer. It uses what happens to be my very first custom shader to render the tileset onto the quad fully on the GPU.

Not all LDtk features are supported yet, but it should work for most simple maps. I'd love to get anybody's feedback on the usage, or the code and implementation.

The biggest issue I don't know how to fix is that while zooming in or out of some maps you end up with some slight bleeding between the tiles related somehow I think to the texture sampling. It ends up taking pixels from the tile next to it on the tilemap. I fixed the large majority of bleeding issues by setting the texture filter mode to `Nearest`, but it still has that issue when zooming in and out sometimes. Maybe it's mipmapping or something. :man_shrugging: 

Anyway, it was lots of fun, and I learned a lot from it. Hopefully somebody else get's some use from it! :smiley:

**Comment by rparrett** [comment]
I'm excited to check this out. I was already planning on moving from `bevy_tiled` to some sort of mashup of `ldtk` and `bevy_tilemap` already, and this might save me some effort!

**Reply by zicklag** [reply]
Awesome! If you have any features that you use that isn't supported yet, please open issues. It will help me figure out what features to prioritize.

**Comment by rparrett** [comment]
I think folks would appreciate this over at `#showcase` on discord, if you're into that sort of thing.

**Reply by cart** [reply]
Agreed. @zicklag should also add it to https://github.com/bevyengine/awesome-bevy

**Reply by zicklag** [reply]
Created a PR for Awesome Bevy. :) 

I'm not on Discord myself, but if anybody else wanted to post it there I'm totally fine with that.

**Comment by zicklag** [comment]
I just pushed an update that fixes 99% of the rendering gliches where the pixels would bleed in between the tiles. There still this 1% situation that I've only seen so far on 1 map that only happens when zooming in and out relatively quickly where there is a line of bleeding pixels for just a brief moment. And it only happens while moving quickly, if you zoom slowly it doesn't happen.

I'm thinking it's got to do with level-of-detail or something else that happens specifically when zooming in and out because it also tends to come with a slight frame rate drop. Anyway, I'm pretty happy with it, now. I'll probably run into the solution for the glitches later. :slightly_smiling_face:
